### ISSUE TITLE
Incorrect escaping of forward slash when JSON typing is applied

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#1687]
While testing release candidate `rc-v0.20.0-suricata16` on macOS, I noticed that the Suricata alert events that contained a forward slash in their Community ID values were not matching up with corresponding Zeek events. Closer inspection reveals what appears to be incorrect escaping, but only when we apply the JSON typing config.

The repro below is with `zq` commit `96bce2c2`.

Here's a single NDJSON event from the EVE log:

```
$ cat one.ndjson 
{"timestamp":"2018-03-23T12:58:26.171395-0700","flow_id":635130576274342,"pcap_cnt":9519,"event_type":"alert","src_ip":"10.47.3.155","src_port":50707,"dest_ip":"134.71.3.16","dest_port":443,"proto":"TCP","community_id":"1:79bZuCMMmkFg6FLG5WIs\/x+8fP0=","tx_id":0,"alert":{"action":"allowed","gid":1,"signature_id":2028369,"rev":2,"signature":"ET JA3 Hash - Possible Malware - Fake Firefox Font Update","category":"Unknown Traffic","severity":3,"metadata":{"updated_at":["2019_10_29"],"former_category":["JA3"],"created_at":["2019_09_10"]}},"app_proto":"tls"}
```

If I look at the Community ID field without applying any typing config:

```
$ zq -t 'cut community_id' one.ndjson 
#0:record[community_id:string]
0:[1:79bZuCMMmkFg6FLG5WIs/x+8fP0=;]
```

If I do the same while applying the Suricata JSON typing config from the zq repo:

```
$ zq -t -j ~/work/zq/ppl/zqd/ingest/suricata-types.json 'cut community_id' one.ndjson 
#0:record[community_id:bstring]
0:[1:79bZuCMMmkFg6FLG5WIs\\/x+8fP0=;]
```

**Comment by henridf** [comment]
@philrz If I change the suricata.yaml `escape-slash` config to `no` (https://suricata.readthedocs.io/en/suricata-6.0.0/output/eve/eve-json-output.html#json-flags), then the community id value in the eve.json log is `1:79bZuCMMmkFg6FLG5WIs/x+8fP0`. Since there's no backslash, both zq json and typed-ndjson (which types the string to bstring) output the same result. Can you confirm that this is what we want, e.g. that it lines up with the zeek-produced community id?

**Comment by philrz** [comment]
Yes, we definitely want the Community ID values from Zeek and Suricata to line up so they can be joined. As you and I discussed offline, neither of us really knows why Suricata does this escaping by default, but it seems undesirable, so I'm keen to disable it to work around the problem and decouple this `zq` JSON typing effect from the Suricata effort.

As for that effect, after closer inspection, the oddity seems to come down to `\/`, which doesn't make much sense to me in JSON, since according to the spec, `/` is not a character that needs to be escaped. If I use `jq` as a reference implementation that does what people would expect to be sane JSON things:

```
$ cat /tmp/uid.ndjson
{"_path":"conn","ts":"2018-03-24T17:15:21.255387Z","uid":"C8Tful1\/TvM3Zf5x8fl"}

$ cat /tmp/uid.ndjson | jq .
{
  "_path": "conn",
  "ts": "2018-03-24T17:15:21.255387Z",
  "uid": "C8Tful1/TvM3Zf5x8fl"
}
```

In other words, they effectively strip the "unnecessary backslash", which is the same thing that `zq` does when we _don't_ apply a JSON typing config.

```
$ zq -f ndjson /tmp/uid.ndjson 
{"_path":"conn","ts":"2018-03-24T17:15:21.255387Z","uid":"C8Tful1/TvM3Zf5x8fl"}

$ zq -f ndjson -j types.json /tmp/uid.ndjson 
{"_path":"conn","_write_ts":null,"conn_state":null,"duration":null,"history":null,"id":{"orig_h":null,"orig_p":null,"resp_h":null,"resp_p":null},"local_orig":null,"local_resp":null,"missed_bytes":null,"orig_bytes":null,"orig_ip_bytes":null,"orig_pkts":null,"proto":null,"resp_bytes":null,"resp_ip_bytes":null,"resp_pkts":null,"service":null,"ts":"2018-03-24T17:15:21.255387Z","tunnel_parents":null,"uid":"C8Tful1\\/TvM3Zf5x8fl"}
```

**Comment by nwt** [comment]
#2587 remove the typed NDJSON reader.

**Comment by philrz** [comment]
While the typed NDJSON reader is gone, on closer inspection I can see that the symptom originally seen as undesirable here started appearing in `zq` when reading any NDJSON data starting with the changes in #2123. Ive opened a new issue #2604 to track this going forward.
